### PR TITLE
Fixes #86: Consider SeqNoAdvanced to fix termination issues caused by unsubscrib…

### LINKED ISSUE #86: generateDatafiles doesn't terminate if the endSeqNo mutation is not from subscribed collections
1. We first find the endSeqno for each vb and use it to close the stream in the future and terminate generateDatafiles step.
2. We also pass a collectionIDs list in OpenStream which signifies that we only need mutations/events from specified collections  to reduce noise and the producer will not stream the unspecified collection mutations/events.
3. This logic causes an issue of untermination of generateDataFiles step if the last N (N > 0) number of mutations in the stream is from the collections we have not subscribed in the OpenStream step and hence producer will not stream them to the consumer. This basically means that generateDatafiles is stuck at (endSeqno - N) sequence number and will never reach endSeqno, causing it to not terminate.

### PR DESCRIPTION
**Pull request by sumukhbhat2701**
- Fixes issue: #86 

This issue is overcome by using DCP_SEQNO_ADV event streamed by the producer which will contain the mutations from a collection that the consumer is not subscribed to.

### CODE REVIEW

**Review by nelio2k — COMMENTED** [review]
Good find. I have two asks:

First, my only concern at this point is to make sure that the whole "CreateMutation" path would not lead to the differ thinking that it is actually an extra mutation on the source when there is not, leading to a false positive.

Is it possible to make sure that there via both double code inspection and also via testing that there won't be a "false positive" case that shows up? Theoretically, I don't think it should but just asking you again for due diligence.

Second, is it possible to run the difftool against the xdcr repository's differ test cases? To make sure that there's no regression. If you can code up a regression test case for this particular instance, please do so and add it to the current set of differ test suites.

**Comment by nelio2k** [comment]
Thirdly, if you can, do a `git commit --amend` and make sure the commit messages follows the format:

```
Fixes #86: consider....
```

This will ensure that once this PR is merged, the issue is closed.

**Review by nelio2k — APPROVED** [review]

**Comment by sumukhbhat2701** [comment]
@nelio2k I don't seem to have the required write access to merge this PR. Can you please help me with this?
Thanks.

**Pull request merged**
